prov/rxm/src/rxm_attr.c

```diff
@@ -260,8 +260,13 @@ static struct fi_info rxm_verbs_info = {
 	.next = &rxm_thru_info,
 };
 
+/* These caps are emulated and/or may degrade performance. */
+#define VALID_OPTIONAL_CAPS_ONLY (FI_DIRECTED_RECV | FI_SOURCE | FI_ATOMIC)
+
 struct util_prov rxm_util_prov = {
 	.prov = &rxm_prov,
 	.info = &rxm_verbs_info,
 	.flags = 0,
+	.opt_caps_valid_caps = ~VALID_OPTIONAL_CAPS_ONLY,
+	.opt_caps_valid_opt_caps = VALID_OPTIONAL_CAPS_ONLY,
```

Comment:
These 2 fields are the same, except for flipping the bits.  Better to keep only opt_caps_valid_opt_caps, but under a more reasonable name, since that directly lists what caps are optional.

Reply:
These two fields serve different purposes:
1. The first defines the optional_caps which can be returned in caps fi_info output
2. The second defines the optional_caps which can be returned in optional_caps fi_info output

By setting both to zero, a provider will not support optional caps. This results in all providers uses the core infrastructure not having optional_caps supported out-of-the-box. Support has to be explicitly added in.

I could change this to the following:
bool optional_caps_enabled: If set, the provider supports optional caps
uint64_t optional_caps_only: If optional_caps is enabled, bit mask of capabilities which can only be returned in optional_caps fi_info output. All other provider capabilities would be eligible to be returned in caps fi_info output.

Thoughts?